kedro/extras/datasets/tensorflow/image_data_generator_dataset.py

```diff
@@ -0,0 +1,79 @@
+import tensorflow as tf 
+from kedro.io.core import AbstractDataSet
+from typing import Any, Dict
+
+class ImageDataGeneratorDataSet(AbstractDataSet):
+
+    def __init__(
+        self,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None
+    ) -> None:
+        """Creates a new instance of `ImageDataGeneratorDataSet`
+            to generate batches of tensor image data with real-time data augmentation.
+            
+            A `tf.keras.preprocessing.ImageDataGenerator` provides methods such as
+            `flow` and `flow_from_directory`. 
+            To use one of these Datasets see `kedro.extras.datasets.tensorflow.FlowFromDirectoryDataSet`
+            See the following for more details of `tf.keras.preprocessing.ImageDataGenerator`:
+            - https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator
+            - https://keras.io/api/preprocessing/image/
+            - https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html
+
+        Args:
+            load_args: Optional arguments passed to `tf.keras.preprocessing.ImageDataGenerator`.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator
+                All defaults are preserved.
+        """
+        self.load_args = load_args
+        self.save_args = save_args 
+
+    def _load(self):
+        return tf.keras.preprocessing.image.ImageDataGenerator(**self.load_args)
+
+    def _save(self, data):
```

Comment:
What about `_exists`? If it's not implemented the assumption is that it will always return False (and log some warning).
You'll also need to implement a `_describe` method.